### PR TITLE
feat: add premium license banner for custom roles page

### LINKED ISSUE #14318: :star: Custom roles no premium license behavior
This feature requires a premium license, not an enterprise license and we should differentiate it somehow.
Custom roles can also still exist, even if the license does not exist, or is expired. Custom roles should still be listed in a view-only mode. So keep the table regardless of the license. Just show an empty table if no custom roles exist.

**Requirements**
- [ ] Theme the premium license in a different color palette and consult Matt V for branding inspiration
- [ ] Paywall should always show if license isn't valid
- [ ] Table should always show regardless of license according to the criteria below
- [ ] Add table for custom roles if custom roles exist even if license is no longer valid
- [ ] Add table for empty state (no custom roles)
- [ ] Add storybooks for different states

Screenshot from 2024-08-15 16-18-06

### PR DESCRIPTION
**Pull request by jaaydenh**
resolves #14318 

- [x]  Theme the premium license in a purple color palette
- [x]  Paywall should always be displayed if license isn't valid
- [x]  Display roles table with any existing custom roles exist even if license is no longer valid
- [x]  Display empty table state  even if license is no longer valid
- [x]  Add storybooks for new states

### CODE REVIEW

**Review by BrunoQuaresma — APPROVED** [review]
LGTM! I did some QA and it works as expected 👍 good job

File: site/src/components/Paywall/Paywall.tsx
Comment (Parkreiner) [inline site/src/components/Paywall/Paywall.tsx:132]: Making a note for myself: we should update our color theme so that it works better on colored backgrounds. The gray divider color looks really washed out on a saturated colored background

**Review by buenos-nachos — APPROVED** [review]
Tentatively approving so that you're not blocked, but I think there's a few changes that can be made to make the component logic more reusable and foolproof

File: site/src/components/Paywall/Paywall.tsx
Comment (Parkreiner) [reply site/src/components/Paywall/Paywall.tsx:132]: Divider thickness doubled to make this more obvious

Before:
Screenshot 2024-09-09 at 10 20 08 AM

After:
Screenshot 2024-09-09 at 10 20 14 AM

File: site/src/theme/branding.ts
Comment (jaaydenh) [reply site/src/theme/branding.ts:7]: No, but I believe the plan going forward is to only have the premium license.

File: site/src/components/Paywall/Paywall.tsx
Comment (jaaydenh) [reply site/src/components/Paywall/Paywall.tsx:116]: got it, removed the marginBottom

File: site/src/components/Paywall/Paywall.tsx
Comment (jaaydenh) [reply site/src/components/Paywall/Paywall.tsx:37]: This was meant to be a temporary solution as I originally thought the enterprise license would only remain for a short time longer. But now I am going to update this to completely remove the enterprise license and leave only the premium

File: site/src/components/Paywall/Paywall.tsx
Comment (aslilac) [reply site/src/components/Paywall/Paywall.tsx:116]: ooooh, purple divider looks _rad_

File: site/src/theme/darkBlue/branding.ts
Comment (aslilac) [inline site/src/theme/darkBlue/branding.ts:18]: you're being specific again :p

the two values that `paywall` and `badge` share–`background` and `border`– are identical. why do they need to be duplicated? I think this should just be flatten. it's not the job of the theme to cater to _specific_ UI components, it's job is to provide general styles.

**Review by aslilac — CHANGES_REQUESTED** [review]

**Review by buenos-nachos — COMMENTED** [review]
I think the content updates look good, but I'm going to defer to Kayla's opinion on the color usage

File: site/src/theme/darkBlue/branding.ts
Comment (jaaydenh) [reply site/src/theme/darkBlue/branding.ts:18]: I still accustomed to other projects where the theme can be specific to UI components, here is an example, https://github.com/supabase/supabase/blob/4dfadb6c03bf832ba875270ccc81f91d278f0df4/packages/ui/build/css/themes/dark.css

I agree here it makes sense to flatten it especially since I removed the enterprise colors.

**Review by aslilac — APPROVED** [review]
thank you! much happier with this 😄

**Pull request merged**
